### ISSUE TITLE
nprogress always show bar

### ISSUE THREAD

**Issue by Sorgrum** [gatsbyjs/gatsby#2975]
I know that it isn't generally needed to show the progress bar all the time. However, small mobile phones the size of the screen can be small enough that header content covers a relatively large portion of the screen. 

While a keen observer may be able to notice a small portion of screen changing, it can be difficult, especially when pages update as quickly as they do when using Gatsby. The problem is expounded by not having any indicator from the browser showing that the action has been completed either.

For my use case I had a small bio at the top of the page on mobile. Without scrolling down you could click on a blog topic and be sent to the full version of the article. But the article also had the small bio at the top of the page and the same excerpt at the top so besides some spacing differences it was hard even for me to notice the difference. I solved my use case by moving the bio to the bottom because it isn't extremely useful, especially on mobile. But I think the progress bar could be useful for clarity purposes. 

Regardless, the Gatsby docs specify "you may pass all available nprogress configuration options." to nprogress plugin. In Gatsby's case it isn't 100% accurate because passing the `minimum` configuration option doesn't change anything in Gatsby's case.

**TL;DR** consider implementing support for the `minimum` configuration option for nprogress instead of the set 1s delayed loading.

**Comment by calcsam** [comment]
Thanks for the feedback. 

Re: the screen size issue, can you include screenshots? 

Re: the accuracy of the README including that sentence, thanks for the point. We'd love a PR fixing this, feel free to submit one!

**Comment by Sorgrum** [comment]
![ambiguity](https://user-images.githubusercontent.com/3497076/33517926-a4050edc-d75a-11e7-9f25-e74f57ebea14.png)

Sorry for the crude drawing. But if you have two different pages, one with the black content, one with the green content, but both have the same top heading. If you change from one page to another without any indicator you might not realize that the page has actually changed because you aren't seeing an actual change in the browser or if you are, it isn't very noticeable. 

So I moved my bio section to the bottom of the page to make the content change more obvious. But having the nprogress could also serve as an indicator that the page has actually changed.

**Comment by fk** [comment]
Hey @Sorgrum!

I agree there are use cases where it makes sense to always show the NProgress loading bar on every page load – no matter if the "[page is delayed in loading](https://github.com/gatsbyjs/gatsby/blob/0c84ed00b22b4c5c5d30a2fe2f1dada503d3b6b7/packages/gatsby-plugin-nprogress/src/gatsby-browser.js#L4)" as per Gatsby's definition ("[one second after clicking on a link](https://www.gatsbyjs.org/packages/gatsby-plugin-nprogress/)") or not. We would need to add an option to the plugin.

---

> Regardless, the Gatsby docs specify "you may pass all available nprogress configuration options." to nprogress plugin. In Gatsby's case it isn't 100% accurate because passing the minimum configuration option doesn't change anything in Gatsby's case.
>
> TL;DR consider implementing support for the minimum configuration option for nprogress instead of the set 1s delayed loading.

What do you expect the NProgress [`minimum`](https://github.com/rstacruz/nprogress#minimum) option to do? How would it help with your issue? NProgress defines that option like this:

> Changes the minimum percentage used upon starting. (default: 0.08)

I quickly set up `gatsby-plugin-nprogress` on www.yisela.com to demonstrate the behavior of NProgress' `minimum` option using the following settings :

```
    {
      resolve: `gatsby-plugin-nprogress`,
      options: {
        color: `#191919`,
        minimum: 0.5,
        trickle: false,
      },
    },
```

If you unregister the service worker, clear the browser cache, then reload the site and shortly after that disable your internet connection, and then try to navigate to one of the articles at the bottom of the page, you should see the NProgress loading bar, filling up 50% of the horizontal space of your browser window (matching `minimum: 0.5`). Because `trickle` is disabled, it should never grow, looking like this:

![image](https://user-images.githubusercontent.com/21834/33521058-9de377f6-d7c7-11e7-9a88-75d74c3e2479.png)

**Comment by Sorgrum** [comment]
Ahh I see what you're saying @fk . I completely misinterpreted the config option.

Seeing as the docs are entirely correct in this case, maybe we implement an addition to the functionality of `gatsby-plugin-nprogress`. 

If instead of 

`gatsby-plugin-nprogress/src/gatsby-browser.js`
```js
window.___emitter.on(`onDelayedLoadPageResources`, () => {
```

if there existed a `window.___emitter.on(`onPreLoadPageResources`, () => {` then an addition would be fairly simple. Although I'm not familiar enough with the source to know if it exists. Maybe someone could provide me with a little guidance and I could submit a PR

**Comment by fk** [comment]
👍  Equally unsure how to approach this 🙄 , but I did a quick search and found [this](https://github.com/gatsbyjs/gatsby/blob/dc4c9a4a09271c64bc7b70da502a1740f2124fb5/packages/gatsby/cache-dir/production-app.js#L76-L82). I'm sure that if @KyleAMathews finds the time he has some input for us. 🙏  😉

**Comment by fk** [comment]
👋  @Sorgrum Should we close this in favor of [#3368 [gatsby-plugin-nprogress]: change default show delay](https://github.com/gatsbyjs/gatsby/issues/3368) which is a little more recent/active?

**Comment by Sorgrum** [comment]
Great idea @fk, even got a response from @KyleAMathews himself. Closing in favor of #3368
